example_plugins/pair_plugin/pytest/test_example_pair.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+from typing import List, Optional, Tuple
+import itertools
+import pytest
+
+import hoomd
+from hoomd.device import Device
+
+from hoomd import pair_plugin
+
+import numpy as np
+from numpy.typing import NDArray
+
+
+def build_binary_system(pos: Optional[List[float]] = None,
+                        device: Optional[Device] = None) -> hoomd.Simulation:
+    """Builds a two-particle system.
+
+    Builds a system of two particles, with one particle located at the origin
+    and the other located at `pos`. `pos` should be a list of 3 floats, and is
+    defaulted to [0.5, 0, 0]. The box is 4x4x4.
+    """
+    if pos is None:
+        pos = [0.5, 0, 0]
+    else:
+        if not isinstance(pos, list):
+            raise ValueError("`pos` must be a list")
+        if len(pos) != 3:
+            raise ValueError("`pos` must be a list of length 3")
+
+    if device is None:
+        device = hoomd.device.CPU()
+
+    sim = hoomd.Simulation(device, 0)
+
+    snap = hoomd.Snapshot()
+    snap.particles.N = 2
+    snap.particles.types = ["A"]
+    snap.particles.position[:] = [[0, 0, 0], pos]
+    snap.configuration.box = [4, 4, 4, 0, 0, 0]
+
+    sim.create_state_from_snapshot(snap)
+
+    return sim
```

Comment:
Could we use the existing `two_particle_snapshot_factory` instead of this method? If not, can we add the necessary optional arguments to `two_particle_snapshot_factory`, or extend it with another fixture?

`two_particle_snapshot_factory` is a pytest fixture, will reuse the existing device fixture, reduce code duplication, and already works with MPI (this method is currently failing MPI tests).
https://github.com/glotzerlab/hoomd-blue/blob/73c79a05f517c1c97b95f92702204c4b7f8405f5/hoomd/conftest.py#L126-L161

Reply:
I just refactored the tests to use the the two snapshot factories like you suggested. The whole `fixture(scope='session')` business was a little surprising to me, but I think some comments will help clarify what's going on to users.